packages/tools/src/store/SynchronizerManager/Synchronizer.ts

```diff
@@ -211,13 +212,20 @@ class Synchronizer {
         if (targetIsSource) {
           continue;
         }
-
-        this._eventHandler(this, sourceViewport, targetViewport, sourceEvent);
+        promises.push(
+          this._eventHandler(this, sourceViewport, targetViewport, sourceEvent)
+        );
       }
     } catch (ex) {
       console.warn(`Synchronizer, for: ${this._eventName}`, ex);
     } finally {
-      this._ignoreFiredEvents = false;
+      if (promises.length) {
```

Comment:
@wayfarer3130  and I reviewed this on a call; this is causing issues in the TVM mode. This is also an important fix that was making the browser hang so we want to keep it.
Mentioning here just so we know we need to fix it before merging

Reply:
Use the viewport status instead of making this async

Reply:
Seems to be ok in the TVM mode now, at least I can't reproduce any hangs.  I believe the issue was caused by multiple initial renders causing a loop, and the existing OHIF changes to prevent extra renders resolved the base issue.

Reply:
@mateusfreira can you try again, It seems to work with tmtv now

Reply:
@mateusfreira  - you have to try it with the merge update branch of FlexView, as it was changes outside of CS3D that fixed the hang issue.

Reply:
@mateusfreira can you try and let me know?